### PR TITLE
Add credential_type module

### PR DESCRIPTION
**Pull request by alinabuzachis**

### CODE REVIEW

**Comment by Akasurde** [comment]
@alinabuzachis Can you please rebase this branch? Thanks.

File: plugins/modules/credential_type.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+
+DOCUMENTATION = """
+---
+module: credential_type
+author:
+  - Alina Buzachis (@alinabuzachis)
+short_description: Manage credential types in EDA Controller
+description:
+  - This module allows to create, update or delete a credential type in EDA controller.
```
Comment (Akasurde) [inline plugins/modules/credential_type.py:19]:
```suggestion
  - This module allows the user to create, update or delete a credential type in the EDA controller.
```

File: plugins/modules/credential_type.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+      - Injectors of the credential type.
+    type: dict
+  state:
+    description:
+      - Desired state of the resource. V(exists) will not modify the resource if it is present.
+    default: "present"
+    choices: ["present", "absent", "exists"]
+    type: str
+requirements:
+  - The 'requests' Python module must be installed.
```
Comment (Akasurde) [inline plugins/modules/credential_type.py:50]:
```suggestion
```

File: plugins/modules/credential_type.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    type: dict
+  injectors:
+    description:
+      - Injectors of the credential type.
+    type: dict
+  state:
+    description:
+      - Desired state of the resource. V(exists) will not modify the resource if it is present.
+    default: "present"
+    choices: ["present", "absent", "exists"]
```
Comment (Akasurde) [inline plugins/modules/credential_type.py:47]: Let us drop state `exists`.

File: plugins/modules/credential_type_info.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  - List credential types in EDA controller.
+version_added: 2.0.0
+options:
+  name:
+    description:
+      - The name of the credential type.
+    type: str
+    required: false
+requirements:
+  - The 'requests' Python module must be installed.
```
Comment (Akasurde) [inline plugins/modules/credential_type_info.py:28]:
```suggestion
```

File: plugins/modules/credential_type_info.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  - name: Get information about a credential type
+    ansible.eda.credential_type_info:
+      name: "Test"
+
+  - name: List all credential types
+    ansible.eda.credential_type_info:
+"""
+
+
+RETURN = """ # """
```
Comment (Akasurde) [inline plugins/modules/credential_type_info.py:44]: Let us populate this information.

File: plugins/modules/credential_type_info.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        timeout=module.params.get("request_timeout"),
+        validate_certs=module.params.get("validate_certs"),
+    )
+
+    name = module.params.get("name")
+    controller = Controller(client, module)
+
+    # Attempt to look up credential_type based on the provided name
+    try:
+        result = controller.get_one_or_many(
```
Comment (Akasurde) [inline plugins/modules/credential_type_info.py:77]: Can you please handle cases where - 
1. Credential type name is provided but none is present 
2. Credential type is returned as dict

```
    if result is None:
        result = []
    if not isinstance(result, list):
        result = [result]
```

File: plugins/modules/credential_type_info.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            "credential-types", name=name, want_one=False
+        )
+    except EDAError as e:
+        module.fail_json(msg=f"Failed to get credential type: {e}")
+
+    module.exit_json(result=result)
```
Comment (Akasurde) [inline plugins/modules/credential_type_info.py:83]: Let us return something meaningful such as `credential_types=result`, what do you think?

**Review by Akasurde — CHANGES_REQUESTED** [review]

**Review by Akasurde — DISMISSED** [review]

**Review by resoluteCoder — APPROVED** [review]

**Review by Akasurde — APPROVED** [review]

**Pull request merged**